### PR TITLE
inlined code global code references support + javadoc link support

### LINKED ISSUE #172: Links with backticks
MDoc displays links with backticks as follows:
screenshot 2018-12-20 at 15 00 01

I think the way IntelliJ displays them is actually nicer:
screenshot 2018-12-20 at 15 00 08

Could we consider restyling this?  The main difference is the underline vs. different background as the highlighting mechanism.

In case it helps, this is the MD that generated the screenshots above:
```
[test link with `backticks`](chapter/page)
```

### PR DESCRIPTION
**Pull request by MykolaGolubyev**
image
image

also closes #172

### CODE REVIEW

**Comment by MykolaGolubyev** [comment]
@jyeakel please review the docs text.
I think with this we are moving very close to making `znai` to handle most of the Java reference documentation cases: javadoc text reuse, references, high level + low level documentation in one. 

I have a ticket to document end-to-end java case. Will try to give it a go soon.

File: znai-docs/documentation/snippets/code-references.md [hunk 1]
```diff
@@ -17,3 +17,12 @@ Add references to `references.csv` if you want all your code snippets to use the
 :include-file: references.csv {title: "references.csv"}
 
 :include-file: global-references.js {title: "API example"}
+
+# Inlined Code
+
+[Inlined code](snippets/inlined-code-snippets) gets automatically converted into a link if its content matches one of the
```
Comment (jyeakel) [inline znai-docs/documentation/snippets/code-references.md]: "is automatically converted"

File: znai-docs/documentation/snippets/java-doc.md [hunk 2]
```diff
@@ -29,6 +29,23 @@ Method level `JavaDoc` text can be referred to as well by specifying the `entry`
     
 :include-java-doc: HelloWorld.java {entry: "numberOfStudents"}
 
+# Handling Links
+
+Links defined with `{@link MyClass}` are automatically converted to [Inlined Code](snippets/inlined-code-snippets) and become
+`MyClass`. To turn `MyClass` into the link you need to use [Code References](snippets/code-references).
```
Comment (jyeakel) [inline znai-docs/documentation/snippets/java-doc.md]: "into a link"

File: znai-docs/documentation/snippets/java-doc.md [hunk 3]
```diff
@@ -29,6 +29,23 @@ Method level `JavaDoc` text can be referred to as well by specifying the `entry`
...
+
+```markdown-and-result
+:include-java-doc: HelloWorld.java {
+    referencesPath: "references/javadoc-references-demo.csv"
+}
+```
+
+:include-file: references/javadoc-references-demo.csv {title: "references/javadoc-references-demo.csv"}
+
+Alternatively you can define [Global References](snippets/code-references#global-references) to turn `MyClass` into link.
```
Comment (jyeakel) [inline znai-docs/documentation/snippets/java-doc.md]: "into a link"

File: znai-docs/documentation/snippets/code-references.md (hunk 1 above)
Comment (jyeakel) [inline znai-docs/documentation/snippets/code-references.md]:
```suggestion
[Inlined code](snippets/inlined-code-snippets) is automatically converted into a link if its content matches one of the
```

File: znai-docs/documentation/snippets/java-doc.md (hunk 2 above)
Comment (jyeakel) [inline znai-docs/documentation/snippets/java-doc.md]:
```suggestion
`MyClass`. To turn `MyClass` into a link you need to use [Code References](snippets/code-references).
```

File: znai-docs/documentation/snippets/java-doc.md (hunk 3 above)
Comment (jyeakel) [inline znai-docs/documentation/snippets/java-doc.md]:
```suggestion
Alternatively you can define [Global References](snippets/code-references#global-references) to turn `MyClass` into a link.
```

File: znai-core/src/main/java/com/twosigma/znai/extensions/file/CodeReferencesTrait.java [hunk 4]
```diff
@@ -21,26 +22,42 @@
...
         this.referencesPath = pluginParams.getOpts().get("referencesPath", null);
         this.referencesFullPath = referencesPath != null ?
                 componentsRegistry.resourceResolver().fullPath(referencesPath):
                 null;
+
+        this.references = buildReferences();
+    }
+
+    public DocReferences getReferences() {
+//        if (references == null) {
```
Comment (MykolaGolubyev) [inline znai-core/src/main/java/com/twosigma/znai/extensions/file/CodeReferencesTrait.java]: removing now

**Pull request merged**